Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamerMSE.cpp

```diff
@@ -701,6 +704,10 @@ void MediaPlayerPrivateGStreamerMSE::updateStates()
                     m_readyState = MediaPlayer::HaveEnoughData;
                     LOG_MEDIA_MESSAGE("m_readyState=%s", dumpReadyState(m_readyState));
                     m_networkState = m_downloadFinished ? MediaPlayer::Idle : MediaPlayer::Loading;
+                } else if (m_mediaSource) {
```

Comment:
I think the entire 'm_buffering' branch can be replaced with one that uses 'm_mediaSource' (in MSE player) and default one can be removed, i.e.:

        case GST_STATE_PAUSED:
        case GST_STATE_PLAYING:
            if (seeking()) {
                m_readyState = MediaPlayer::HaveMetadata;
                // TODO: NetworkState?
                LOG_MEDIA_MESSAGE("m_readyState=%s", dumpReadyState(m_readyState));
            } else if (m_downloadFinished) {
                m_readyState = MediaPlayer::HaveEnoughData;
                LOG_MEDIA_MESSAGE("m_readyState=%s", dumpReadyState(m_readyState));
                m_networkState = MediaPlayer::Loaded;
            } else if (m_mediaSource) {
                m_mediaSource->monitorSourceBuffers();
                LOG_MEDIA_MESSAGE("m_readyState=%s", dumpReadyState(m_readyState));
                m_networkState = MediaPlayer::Loading;
            }
            break;

does it look OK for you?